src/components/Detail/Detail.tsx

```diff
@@ -0,0 +1,99 @@
+import { useRouter } from 'next/router';
+import classnames from 'classnames';
+
+import { Row } from '../Row/Row';
+import { trpc } from '../../utils/trpc';
+import style from './Detail.module.scss';
+
+export const DetailOfActivity = () => {
+  const router = useRouter();
+  const { id } = router.query;
+
+  const utils = trpc.useContext();
+
+  const { data: activityData } = trpc.activities.read.useQuery({
+    id: Number(id),
+  });
+
+  const deleteActivity = trpc.activities.delete.useMutation({
+    onSuccess: () => {
+      utils.activities.invalidate();
+    },
+  });
+
+  const handleDelete = (activityId: number) => {
+    deleteActivity.mutate({ id: activityId });
+
+    router.push('/');
+  };
+
+  const updateDoneCounter = trpc.activities.updateDoneCounter.useMutation({
+    onSuccess: () => {
+      utils.activities.invalidate();
+    },
+  });
+
+  const handleActivityClick = ({
+    activityId,
+    timesDone,
+  }: {
+    activityId: number;
+    timesDone: number;
+  }) => {
+    updateDoneCounter.mutate({ id: activityId, timesDone: timesDone + 1 });
+  };
+
+  const days = (timesDone: number) => {
+    if (timesDone === 1) {
+      return 'den';
+    }
+    if (timesDone > 1 && timesDone < 5) {
+      return 'dny';
+    }
+
+    return 'dní';
+  };
+
+  return (
+    activityData && (
```

Comment:
When some component has top-level conditional rendering like this. You should strongly prefer an early return. Avoid ternaries, where the falsy branch is somewhere a million lines down the JSX. Avoid this top-level operand selector as well.

```
if(!activityData) {
 return null // <- When the component does not return anything, it returns null. That is something to remember as well
}
```

This pattern implies functions as well. Try to avoid nesting of IFs in favor of early returns.

- https://www.digitalocean.com/community/posts/5-tips-to-write-better-conditionals-in-javascript#2-less-nesting-return-early
- https://www.codingbeautydev.com/blog/stop-using-nested-ifs